lib/sparse-stream/shared.ts

```diff
@@ -120,7 +108,7 @@ function verifyOrGenerateChecksum(
 		const checksum = hasher.digest('hex') as string;
 		if (generateChecksums) {
 			blocks.checksum = checksum;
-		} else if (verify && blocks.checksum !== checksum) {
+		} else if (verify && blocks.checksum?.toString() !== checksum.toString()) {
```

Comment:
`blocks.checksum` is either `undefined` or a string, no need to call `.toString()` on it;
`checksum` is supposed to be a string to, at least according to the `as string above`

Reply:
@zvin Well, I think with the changes, is can either be a buffer or a string...I am not sure why this happened, but I will look into it.